pkg/scheduler/core/generic_scheduler.go

```diff
@@ -388,7 +404,18 @@ func addNominatedPods(podPriority int32, meta algorithm.PredicateMetadata,
 	return true, metaOut, nodeInfoOut
 }
 
+// equivalenceClassInfo holds information of equivalence cache for specific pod.
+// We will pass this to podFitsOnNode to ensure equivalence hash is only calculated per schedule.
+type equivalenceClassInfo struct {
```

Comment:
Could you please move this to `equivalence_cache.go` and change `getHashEquivalencePod` to return a pointer to this struct?

Reply:
Fixed. In this case, we can use `getHashEquivalencePod()` return nil or not as available flag.